corehq/motech/generic_inbound/views.py

```diff
@@ -0,0 +1,158 @@
+        return ConfigurableAPI.objects.filter(domain=self.domain)
+
+    @property
+    def total(self):
+        return self.base_query.count()
+
+    def post(self, *args, **kwargs):
+        return self.paginate_crud_response
+
+    @property
+    def column_names(self):
+        return [
+            _("Name"),
+            _("Description"),
+            _("URL"),
+            _("Actions"),
+        ]
+
+    @property
+    def page_context(self):
+        return self.pagination_context
+
+    @property
+    def paginated_list(self):
+        for api in self.base_query.all():
+            yield {
+                "itemData": self._item_data(api),
+                "template": "base-api-config-template",
+            }
+
+    def _item_data(self, api):
+        return {
+            'id': api.id,
+            'name': api.name,
+            'description': api.description,
+            'api_url': api.absolute_url,
+            'edit_url': reverse(ConfigurableAPIEditView.urlname, args=[self.domain, api.id]),
+        }
+
+    def get_create_form(self, is_blank=False):
+        if self.request.method == 'POST' and not is_blank:
+            return ConfigurableAPICreateForm(self.request, self.request.POST)
+        return ConfigurableAPICreateForm(self.request)
+
+    def get_create_item_data(self, create_form):
+        new_api = create_form.save()
+        return {
+            "itemData": self._item_data(new_api),
+            "template": "base-api-config-template",
+        }
+
+
+@method_decorator(toggles.GENERIC_INBOUND_API.required_decorator(), name='dispatch')
+class ConfigurableAPIEditView(BaseProjectSettingsView):
+    page_title = gettext_lazy("Edit API Configuration")
+    urlname = "configurable_api_edit"
+    template_name = "generic_inbound/api_edit.html"
+
+    @property
+    def api_id(self):
+        return self.kwargs['api_id']
+
+    @property
+    @memoized
+    def api(self):
+        try:
+            return ConfigurableAPI.objects.get(id=self.api_id)
+        except ConfigurableAPI.DoesNotExist:
+            raise Http404
+
+    @property
+    def page_url(self):
+        return reverse(self.urlname, args=(self.domain, self.api_id,))
+
+    def get_form(self):
+        if self.request.method == 'POST':
+            return ConfigurableAPIUpdateForm(self.request, self.request.POST, instance=self.api)
+        return ConfigurableAPIUpdateForm(self.request, instance=self.api)
+
+    @property
+    def main_context(self):
+        main_context = super(ConfigurableAPIEditView, self).main_context
+
+        main_context.update({
+            "form": self.get_form(),
+            "api_model": self.api,
+        })
+        return main_context
+
+    def post(self, request, domain, **kwargs):
+        form = self.get_form()
+        if form.is_valid():
+            form.save()
+            messages.success(request, _("API Configuration updated successfully."))
+        return redirect(self.page_url, domain, self.api_id)
```

Comment:
I think to render the bound form with errors when it's invalid, it needs to be:
```suggestion
            return redirect(self.page_url, domain, self.api_id)
        return self.get(request, domain, **kwargs)
```
It's been a while since I've done this though, so I could be mistaken

Reply:
d8477568202